### PR TITLE
Add layer with wards, LADs, and CAs. #285

### PR DESCRIPTION
**Pull request by dabreegster**
Thanks to Alex for pointing us to the data source! The import is in https://github.com/acteng/atip-data-prep/pull/17.

Demo: https://acteng.github.io/atip/wards/browse.html

### CODE REVIEW

File: src/lib/browse/ParliamentaryConstituenciesLayerControl.svelte [hunk 1]
```diff
@@ -62,6 +62,7 @@
     let name = e.detail.properties.Name;
     name = name.replace(/ Boro Const$/, "");
     name = name.replace(/ Co Const$/, "");
+    name = encodeURIComponent(name);
```
Comment (dabreegster) [inline src/lib/browse/ParliamentaryConstituenciesLayerControl.svelte:65]: Need to be paranoid about "&" and things

File: src/lib/browse/WardsLayerControl.svelte [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+  }
+
+  function tooltip(feature: MapGeoJSONFeature): string {
+    return `<p>${feature.properties.name}</p>`;
+  }
+
+  function onClick(e: CustomEvent<MapGeoJSONFeature>) {
+    let name = encodeURIComponent(e.detail.properties.name);
+    // Help people find the councillor for this area
```
Comment (dabreegster) [inline src/lib/browse/WardsLayerControl.svelte:62]: I could not do any better than Google this time :(

File: src/lib/browse/colors.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export const colors = {
   hospitals: "#B73D25",
   mrn: "#006478",
   parliamentary_constituencies: "#006E59",
+  wards: "purple",
```
Comment (dabreegster) [inline src/lib/browse/colors.ts:7]: Back to just making these up. We need a principled way of doing this.

**Review by dabreegster — COMMENTED** [review]
UX problems when there are multiple boundaries turned on, but we need to know how people want to use this before figuring out a fix. (Like only allowing one boundary layer at a time)

**Comment by robinlovelace-ate** [comment]
My ward is in there and up-to-date. 👍 on that basis, like the link to Google for further information, also not sure if we can do better than that... would hope so!

image

**Comment by robinlovelace-ate** [comment]
Broader thought probably already discussed but just while in my head: easy to add these layers to the scheme editing and other pages at some point in the future?

File: src/lib/browse/CombinedAuthoritiesLayerControl.svelte
Comment (Pete-Y-CS) [inline src/lib/browse/CombinedAuthoritiesLayerControl.svelte:1]: As I think we've discussed, a lot of these controls share stuff in common, maybe we should log an issue to tidy them up once we've created all the layers that exist and we can think clearly about what can be shared?

File: src/lib/browse/colors.ts (hunk 3 above)
Comment (Pete-Y-CS) [reply src/lib/browse/colors.ts:7]: I think we're also gonna run out of significantly distinct colours.

**Review by Pete-Y-CS — APPROVED** [review]

File: src/lib/browse/colors.ts (hunk 3 above)
Comment (dabreegster) [reply src/lib/browse/colors.ts:7]: We could maybe make all the boundary layers one color, and effectively change this to a radio button. But I'm not sure -- we need to know how people want to use this.

File: src/lib/browse/CombinedAuthoritiesLayerControl.svelte
Comment (dabreegster) [reply src/lib/browse/CombinedAuthoritiesLayerControl.svelte:1]: Yep, 100% agree. I think we need a few more first to understand commonalities. The census ones are turning out to be special in a few ways...

File: src/lib/browse/colors.ts (hunk 3 above)
Comment (Pete-Y-CS) [reply src/lib/browse/colors.ts:7]: Yeah, I also imagined we could have a set of colours which got applied to things as you ticked em, and you are only allowed a certain number of layers at once. A changing legend does sound like bad UX though so probably not a good idea

**Pull request merged**